notification/app/notification/models/azure/Notification.scala

```diff
@@ -0,0 +1,98 @@
+package notification.models.azure
+
+import java.net.URL
+import java.util.UUID
+import azure.{Tag, AzureRawPush}
+import models.{UserId, Push, GoalType, Topic}
+import play.api.libs.json._
+import models.JsonUtils._
+
+sealed trait Notification {
+  def id: UUID
+  def `type`: String
+  def title: String
+  def message: String
+  def thumbnailUrl: Option[URL]
+  def topic: Set[Topic]
+}
+
+object Notification {
+
+  def toAzureRawPush(push: Push) = {
+    val body = Json.stringify(Json.toJson(push.notification))
+    push.destination match {
+      case Left(topic: Topic) => AzureRawPush("wns/raw", body, Some(Set(Tag.fromTopic(topic))))
```

Comment:
"wns/raw" is default value in constructor
BTW: why wnsType is exposed as ctor parameter ?